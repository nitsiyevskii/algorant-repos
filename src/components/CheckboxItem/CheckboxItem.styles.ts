import { createStyles } from 'src/utils/createStyles';

export const useStyles = createStyles((theme) => ({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: 12,
  },
  checkbox: {
    width: 24,
    height: 24,
    borderRadius: 6,
    borderWidth: 2,
    borderColor: theme.colors.core.primary,
    marginRight: 12,
    alignItems: 'center',
    justifyContent: 'center',
  },
  label: {
    fontSize: theme.fonts.size.md,
    color: theme.colors.text.primary,
  },
}));

