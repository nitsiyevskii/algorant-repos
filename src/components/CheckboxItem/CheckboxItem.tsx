import React from 'react';
import { TouchableOpacity, View, Text } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { useTheme } from 'src/hooks/useTheme';
import { useStyles } from './CheckboxItem.styles';

interface CheckboxItemProps {
  label: string;
  isSelected: boolean;
  onPress: () => void;
  testID?: string;
}

export const CheckboxItem: React.FC<CheckboxItemProps> = ({
  label,
  isSelected,
  onPress,
  testID,
}) => {
  const theme = useTheme();
  const styles = useStyles();

  return (
    <TouchableOpacity
      style={styles.container}
      onPress={onPress}
      testID={testID}
    >
      <View style={styles.checkbox}>
        {isSelected && (
          <Ionicons name="checkmark" size={18} color={theme.colors.core.primary} />
        )}
      </View>
      <Text style={styles.label}>{label}</Text>
    </TouchableOpacity>
  );
};

