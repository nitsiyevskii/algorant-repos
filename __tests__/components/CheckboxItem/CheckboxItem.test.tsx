import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { CheckboxItem } from 'src/components/CheckboxItem/CheckboxItem';

describe('CheckboxItem', () => {
  const mockOnPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with label', () => {
    const { getByText } = render(
      <CheckboxItem label="Option 1" isSelected={false} onPress={mockOnPress} />
    );
    expect(getByText('Option 1')).toBeTruthy();
  });

  it('renders as selected', () => {
    const { toJSON } = render(
      <CheckboxItem label="Option 2" isSelected={true} onPress={mockOnPress} />
    );
    expect(toJSON()).toBeTruthy();
  });

  it('renders as unselected', () => {
    const { toJSON } = render(
      <CheckboxItem label="Option 3" isSelected={false} onPress={mockOnPress} />
    );
    expect(toJSON()).toBeTruthy();
  });

  it('calls onPress when pressed', () => {
    const { getByTestId } = render(
      <CheckboxItem
        label="Option 4"
        isSelected={false}
        onPress={mockOnPress}
        testID="checkbox-option4"
      />
    );

    const checkbox = getByTestId('checkbox-option4');
    fireEvent.press(checkbox);

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot when unselected', () => {
    const { toJSON } = render(
      <CheckboxItem label="Algorand" isSelected={false} onPress={mockOnPress} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot when selected', () => {
    const { toJSON } = render(
      <CheckboxItem label="Pera Wallet" isSelected={true} onPress={mockOnPress} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});

