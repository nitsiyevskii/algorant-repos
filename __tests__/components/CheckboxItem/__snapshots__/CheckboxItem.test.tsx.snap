// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxItem matches snapshot when selected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
      "paddingVertical": 12,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#007AFF",
        "borderRadius": 6,
        "borderWidth": 2,
        "height": 24,
        "justifyContent": "center",
        "marginRight": 12,
        "width": 24,
      }
    }
  >
    <Text
      color="#007AFF"
      name="checkmark"
      size={18}
      testID="icon"
    >
      checkmark
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 16,
      }
    }
  >
    Pera Wallet
  </Text>
</View>
`;

exports[`CheckboxItem matches snapshot when unselected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
      "paddingVertical": 12,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#007AFF",
        "borderRadius": 6,
        "borderWidth": 2,
        "height": 24,
        "justifyContent": "center",
        "marginRight": 12,
        "width": 24,
      }
    }
  />
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 16,
      }
    }
  >
    Algorand
  </Text>
</View>
`;
